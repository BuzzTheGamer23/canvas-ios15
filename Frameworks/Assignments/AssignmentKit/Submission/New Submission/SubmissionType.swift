//
//  SubmissionType.swift
//  Assignments
//
//  Created by Nathan Armstrong on 1/23/17.
//  Copyright Â© 2017 Instructure. All rights reserved.
//

public enum SubmissionType {
    case text
    case url
    case fileUpload

    var title: String {
        switch self {
        case .text:
            return NSLocalizedString("Enter Text",
                                     tableName: "Localizable",
                                     bundle: .assignments(),
                                     value: "",
                                     comment: "Text submission option")
        case .url:
            return NSLocalizedString("Add Website Address",
                                     tableName: "Localizable",
                                     bundle: .assignments(),
                                     value: "",
                                     comment: "URL submission option")
        case .fileUpload:
            return NSLocalizedString("Upload File",
                                     tableName: "Localizable",
                                     bundle: .assignments(),
                                     value: "",
                                     comment: "File upload submission option")
        }
    }
}
