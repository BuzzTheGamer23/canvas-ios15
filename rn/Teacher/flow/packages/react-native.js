// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
