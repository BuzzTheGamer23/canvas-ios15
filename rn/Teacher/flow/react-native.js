//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
