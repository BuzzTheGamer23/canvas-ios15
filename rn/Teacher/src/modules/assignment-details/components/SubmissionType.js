/**
 * @flow
 */

import React, { Component } from 'react'
import { Text } from '../../../common/text'
import { SUBMISSION_TYPES } from '../../../common/submissionTypes'

import {
  View,
  StyleSheet,
} from 'react-native'

export default class SubmissionType extends Component {
  render () {
    return (<View style={style.vertical}>
      {this.props.data.map(item =>
        <Text style={{ flex: 1 }} key={item}>{SUBMISSION_TYPES[item]}</Text>
      )}
    </View>)
  }
}

const style = StyleSheet.create({
  vertical: {
    flexDirection: 'column',
  },
})
