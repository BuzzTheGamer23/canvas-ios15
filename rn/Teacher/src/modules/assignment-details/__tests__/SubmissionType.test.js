/**
 * @flow
 */

import 'react-native'
import React from 'react'
import SubmissionType from '../components/SubmissionType'
import renderer from 'react-test-renderer'

const defaultProps = {
  data: ['foo', 'bar'],
}

test('render', () => {
  let tree = renderer.create(
    <SubmissionType {...defaultProps} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render 0 types', () => {
  defaultProps.data = []
  let tree = renderer.create(
    <SubmissionType {...defaultProps} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render single type', () => {
  defaultProps.data = ['foo']
  let tree = renderer.create(
    <SubmissionType {...defaultProps} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
