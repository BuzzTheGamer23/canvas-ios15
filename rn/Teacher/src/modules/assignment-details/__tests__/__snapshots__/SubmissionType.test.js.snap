// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 0 types 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
/>
`;

exports[`render 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
        },
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
        },
      ]
    }
  />
</View>
`;

exports[`render single type 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
        },
      ]
    }
  />
</View>
`;
